cosmos/azure-spring-data-cosmos/cosmos-multi-tenant-samples/tenant-by-container/src/main/java/com/azure/spring/data/cosmos/example/tenant/MultiTenantContainerCosmosFactory.java

```diff
@@ -0,0 +1,45 @@
+package com.azure.spring.data.cosmos.example.tenant;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.models.ThroughputProperties;
+import com.azure.spring.data.cosmos.CosmosFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+
+public class MultiTenantContainerCosmosFactory extends CosmosFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(MultiTenantContainerCosmosFactory.class);
+    private CosmosAsyncClient client;
+    private Environment env;
+    @Autowired
+    private TenantStorage tenantStorage;
+    public String tenantId;
+
+    public MultiTenantContainerCosmosFactory(CosmosAsyncClient cosmosAsyncClient, String databaseName, Environment env) {
+        super(cosmosAsyncClient, databaseName);
+        this.client = cosmosAsyncClient;
+        this.env = env;
+        this.tenantId = databaseName;
```

Comment:
This could be confusing, this is a tenant by container example but we are setting tenantId to databaseName? Maybe we should rename this field?

Reply:
Actually on second thought is this even the desired setting, or should we be setting this to a container name here?

Reply:
yes confusing, should be in the container method, will remove